cubeviz/tools/collapse_cube.py

```diff
@@ -311,187 +228,321 @@ def hide_start_end(self, dohide):
-            self.error_label_text.setText('Can not have both start and end outside of the wavelength range.')
-            return
-
-        if start_index > end_index:
-            self.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
-            self.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
-            self.error_label_text.setText('Start value must be less than end value')
+            start_index, end_index = self._calculate_callback_index_checks(start_value, end_value)
+            
+        if  start_index is None or end_index is None:
             return
 
-
         # Check to see if the wavelength (indices) are the same.
         if start_index == end_index:
-            self.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
-            self.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
-            self.error_label_text.setText('Can not have both start and end wavelengths be the same.')
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Can not have both start and end wavelengths be the same.')
+            self.ui.error_label.setVisible(True)
             return
 
         # Set the start and end values in the text boxes -- in case they enter one way out of range then
         # we'll fix it.
-        ts = start_index if 'Indices' in self.region_combobox.currentText() else self.wavelengths[start_index]
-        self.start_text.setText('{}'.format(ts))
-
-        te = end_index if 'Indices' in self.region_combobox.currentText() else self.wavelengths[end_index]
-        self.end_text.setText('{}'.format(te))
+#        ts = start_index if 'Indices' in self.region_combobox.currentText() else self.wavelengths[start_index]
+#        self.ui.start_input.setText('{}'.format(ts))
+#
+#        te = end_index if 'Indices' in self.region_combobox.currentText() else self.wavelengths[end_index]
+#        self.ui.end_input.setText('{}'.format(te))
 
 
         data_name = self.data_combobox.currentText()
         operation = self.operation_combobox.currentText()
 
-        # Do calculation if we got this far
-        new_wavelengths, new_component = collapse_cube(self.data[data_name], data_name, self.data.coords.wcs,
-                                             operation, start_index, end_index)
-
         # Get the start and end wavelengths from the newly created spectral cube and use for labeling the cube.
         # Convert to the current units.
-        start_wavelength = new_wavelengths[0].to(self.wavelength_units)
-        end_wavelength = new_wavelengths[-1].to(self.wavelength_units)
+        start_wavelength = self.wavelengths[start_index]
+        end_wavelength = self.wavelengths[end_index]
 
-        label = '{}-collapse-{} ({:0.3}, {:0.3})'.format(data_name, operation,
+        label = '{}-collapse-{} ({:0.4}, {:0.4})'.format(data_name, operation,
```

Comment:
There is currently not enough precision in the wavelength values to prevent duplicated data components.

Reply:
I'm not sure this fixes the issue in a really robust way...